ribbon-eureka/src/main/java/com/netflix/niws/loadbalancer/Eureka2EnabledNIWSServerList.java

```diff
@@ -0,0 +1,211 @@
+    private boolean shouldUseIpAddr;
+
+    private AtomicReference<ServerListReader> serverListReaderRef = new AtomicReference<ServerListReader>();
+
+    @Override
+    public void initWithNiwsConfig(IClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+        clientName = clientConfig.getClientName();
+        vipAddresses = clientConfig.resolveDeploymentContextbasedVipAddresses();
+        if (vipAddresses == null &&
+                ConfigurationManager.getConfigInstance().getBoolean("DiscoveryEnabledNIWSServerList.failFastOnNullVip", true)) {
+            throw new NullPointerException("VIP address for client " + clientName + " is null");
+        }
+        isSecure = clientConfig.getPropertyAsBoolean(CommonClientConfigKey.IsSecure, false);
+        prioritizeVipAddressBasedServers = clientConfig.getPropertyAsBoolean(CommonClientConfigKey.PrioritizeVipAddressBasedServers, prioritizeVipAddressBasedServers);
+        datacenter = ConfigurationManager.getDeploymentContext().getDeploymentDatacenter();
+
+        shouldUseIpAddr = clientConfig.getPropertyAsBoolean(CommonClientConfigKey.UseIPAddrForServer, DefaultClientConfigImpl.DEFAULT_USEIPADDRESS_FOR_SERVER);
+
+        // override client configuration and use client-defined port
+        if (clientConfig.getPropertyAsBoolean(CommonClientConfigKey.ForceClientPortConfiguration, false)) {
+            if (isSecure) {
+                if (clientConfig.containsProperty(CommonClientConfigKey.SecurePort)) {
+                    overridePort = clientConfig.getPropertyAsInteger(CommonClientConfigKey.SecurePort, DefaultClientConfigImpl.DEFAULT_PORT);
+                    shouldUseOverridePort = true;
+                } else {
+                    logger.warn(clientName + " set to force client port but no secure port is set, so ignoring");
+                }
+            } else {
+                if (clientConfig.containsProperty(CommonClientConfigKey.Port)) {
+                    overridePort = clientConfig.getPropertyAsInteger(CommonClientConfigKey.Port, DefaultClientConfigImpl.DEFAULT_PORT);
+                    shouldUseOverridePort = true;
+                } else {
+                    logger.warn(clientName + " set to force client port but no port is set, so ignoring");
+                }
+            }
+        }
+    }
+
+    @Override
+    public List<DiscoveryEnabledServer> getInitialListOfServers() {
+        return obtainServersViaEureka2();
+    }
+
+    @Override
+    public List<DiscoveryEnabledServer> getUpdatedListOfServers() {
+        return obtainServersViaEureka2();
+    }
+
+    /**
+     * TODO prioritizeVipAddressBasedServers not supported yet; all vips are subscribed to eagerly
+     */
+    private List<DiscoveryEnabledServer> obtainServersViaEureka2() {
+        logger.info("Resolving {} from Eureka2...", this.vipAddresses);
+        List<InstanceInfo> instanceInfos = getServerListReader().getLatestServerListOrWait();
```

Comment:
On a second thought I will make it configurable via property. Some tuning should be possible.